consensus/qbft/src/test/java/org/hyperledger/besu/consensus/qbft/QbftBlockHeaderValidationRulesetFactoryTest.java

```diff
@@ -63,7 +63,7 @@ public void bftValidateHeaderPasses() {
         getPresetHeaderBuilder(2, proposerNodeKey, validators, parentHeader).buildHeader();
 
     final BlockHeaderValidator validator =
-        QbftBlockHeaderValidationRulesetFactory.blockHeaderValidator(5).build();
+        new QbftBlockHeaderValidationRulesetFactory(false).blockHeaderValidator(5).build();
```

Comment:
Can some tests be added for when we are using contract based validation rules

Reply:
added couple of new test classes to cover new logic.